### PR TITLE
SL-2198 - CLI --output option

### PR DESCRIPTION
**Pull request by philsturgeon**
### Please check if the PR fulfills these requirements 

- [ ] Tests for the changes have been added (for bug fixes / features) 
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?

New feature

### What is the current behavior?

Output from CLI is always sent to stdout. Piping all output results in some unwanted content such as progress reports.

### If this is a feature change, what is the new behavior?

Adds a --option flag to the CLI.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,16 +21,31 @@
...
+    "plugins": [
+      "@oclif/plugin-help"
+    ]
+  },
   "files": [
-    "**/*"
+    "./*",
+    "src/*",
+    "src/cli/commands",
+    "src/functions",
```
Comment (marbemac) [inline package.json]: Should stick with `**/*` for files array.

File: package.json [hunk 2]
```diff
@@ -39,28 +54,43 @@
...
     "test": "jest",
-    "test.prod": "yarn lint && yarn test --coverage --maxWorkers=2",
+    "test.prod": "yarn build && yarn lint && yarn test --coverage --maxWorkers=2",
     "test.update": "yarn test --updateSnapshot",
     "test.watch": "yarn test --watch"
   },
   "dependencies": {
+    "@oclif/command": "^1",
+    "@oclif/plugin-help": "^2",
+    "@stoplight/json-ref-resolver": "^1.2",
```
Comment (marbemac) [inline package.json]: Suggest pegging stoplight deps to major/minor e.g. `1.2.x` (same for json and types).

File: src/cli/commands/lint.ts [hunk 3]
```diff
@@ -93,20 +98,35 @@ async function lint(name: string, flags: any, command: Lint) {
...
-  }[format]();
+  }[flags.format]();
+}
+
+async function writeOutput(outputStr: string, flags: any, command: Lint) {
+  if (flags.output) {
+    const writeFileAsync = promisify(writeFile);
+    return writeFileAsync(flags.output, outputStr);
+  } else {
+    command.log(outputStr);
```
Comment (P0lip) [inline src/cli/commands/lint.ts]:
```suggestion
```

File: src/cli/commands/lint.ts [hunk 4]
```diff
@@ -93,20 +98,35 @@ async function lint(name: string, flags: any, command: Lint) {
...
+    return Promise.resolve();
```
Comment (P0lip) [inline src/cli/commands/lint.ts]:
```suggestion
    return;
```

**Pull request merged**